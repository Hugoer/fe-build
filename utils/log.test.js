const { severity, log } = require("./log");

describe('Test utils/log.js', () => {
    Object.keys(severity).forEach((severetyName,index) => {
    it(`Check formating of each error severety send to console log [${severetyName}]`, () => {
            const fn = severetyName === 'error' ? severetyName : 'log';
            const file = __filename;
            const title = 'My error';
            const extra = 'Extra error information';
            const text = 'my error';
            console[fn] = jest.fn();
            log(file, title, extra, severetyName);
            expect(console[fn].mock.calls[0][0]).toMatchSnapshot();
        });
    });
});

