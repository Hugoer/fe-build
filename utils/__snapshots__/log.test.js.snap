// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test utils/log.js Check formating of each error severety send to console log [error] 1`] = `"[2m[utils/log.test.js] [0m[31mâœ–  [0m[31mMy error[0m[2mExtra error information[0m"`;

exports[`Test utils/log.js Check formating of each error severety send to console log [info] 1`] = `"[2m[utils/log.test.js] [0m[2mâ˜ž  [0m[2mMy error[0m[2mExtra error information[0m"`;

exports[`Test utils/log.js Check formating of each error severety send to console log [log] 1`] = `"[2m[utils/log.test.js] [0m[33mâš   [0m[37mMy error[0m[2mExtra error information[0m"`;

exports[`Test utils/log.js Check formating of each error severety send to console log [success] 1`] = `"[2m[utils/log.test.js] [0m[2m ðŸ’ª  -> [0m[32mMy error[0m[2mExtra error information[0m"`;

exports[`Test utils/log.js Check formating of each error severety send to console log [warning] 1`] = `"[2m[utils/log.test.js] [0m[33mâš   [0m[33mMy error[0m[2mExtra error information[0m"`;
